### PR TITLE
[TAS-2289] 📈 Add crisp user segment from server purchase events

### PR DESCRIPTION
**Pull request by williamchong**

### CODE REVIEW

File: src/util/crisp.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        nickname: displayName || wallet || email.split('@')[0],
+      },
+      segments,
+      active: Math.floor(Date.now() / 1000),
+    });
+  }
+  await CrispClient.website.updatePeopleData(CRISP_WEBSITE_ID, email, {
+    like_wallet: wallet,
+    login_method: loginMethod,
+  });
```
Comment (nwingt) [inline src/util/crisp.ts:69]: Why do we call this separately?
Can this be merged with the `addPeopleData`/`updatePeopleData` above?

File: src/util/api/likernft/book/purchase.ts [hunk 2]
```diff
@@ -1378,6 +1380,20 @@ export async function processNFTBookStripePurchase(
         stripeFeeAmount,
       }),
     ]);
+
+    if (email) {
+      const segments = ['purchaser'];
+      if (feeInfo.customPriceDiff) segments.push('tipper');
+      if (CRISP_WALLET_TO_SEGMENT_MAPPING?.[ownerWallet]) {
+        segments.push(CRISP_WALLET_TO_SEGMENT_MAPPING[ownerWallet]);
+      }
```
Comment (nwingt) [inline src/util/api/likernft/book/purchase.ts]: it seems that we can refactor this part

Comment (williamchong) [reply src/util/api/likernft/book/purchase.ts]: You mean wrap these three lines as a function in util/crisp?

Comment (nwingt) [reply src/util/api/likernft/book/purchase.ts]: Yes

**Review by nwingt — APPROVED** [review]

**Pull request merged**